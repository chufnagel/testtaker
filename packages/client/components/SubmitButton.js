import Link from "next/link";
import PropTypes from "prop-types";

const SubmitButton = ({ path, query, action, text }) => (
  <div style={{ marginLeft: 130 }}>
    <Link href={{ pathname: path, query }}>
      <button onClick={action} type="button">
        {text}
      </button>
    </Link>
  </div>
);

SubmitButton.propTypes = {
  action: PropTypes.func.isRequired,
  path: PropTypes.string.isRequired,
  query: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
};

export default SubmitButton;
