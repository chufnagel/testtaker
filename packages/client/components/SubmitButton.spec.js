/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import SubmitButton from "./SubmitButton";

// SubmitButton.propTypes = {
//   action: PropTypes.func.isRequired,
//   path: PropTypes.string.isRequired,
//   query: PropTypes.string.isRequired,
// };

const mockFn = jest.fn();

describe("SubmitButton", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SubmitButton action={mockFn} text="Submit" path="/" query={{}} />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("SubmitButton matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
