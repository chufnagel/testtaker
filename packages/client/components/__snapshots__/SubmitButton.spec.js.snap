// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton SubmitButton matches snapshot 1`] = `
<div
  style={
    Object {
      "marginLeft": 130,
    }
  }
>
  <Link
    href={
      Object {
        "pathname": "/",
        "query": Object {},
      }
    }
  >
    <button
      onClick={[MockFunction]}
      type="button"
    >
      Submit
    </button>
  </Link>
</div>
`;
